src/transformers/modeling_attn_mask_utils.py

```diff
@@ -199,80 +200,11 @@ def _unmask_unattended(
-
-        For example, if `attention_mask` is
-        ```
-        [[0, 0, 1],
-         [1, 1, 1],
-         [0, 1, 1]]
-        ```
-        and `expanded_mask` is (e.g. here left-padding case)
-        ```
-        [[[[0, 0, 0],
-           [0, 0, 0],
-           [0, 0, 1]]],
-         [[[1, 0, 0],
-           [1, 1, 0],
-           [1, 1, 1]]],
-         [[[0, 0, 0],
-           [0, 1, 0],
-           [0, 1, 1]]]]
-        ```
-        then the modified `expanded_mask` will be
-        ```
-        [[[[1, 1, 1],   <-- modified
-           [1, 1, 1],   <-- modified
-           [0, 0, 1]]],
-         [[[1, 0, 0],
-           [1, 1, 0],
-           [1, 1, 1]]],
-         [[[1, 1, 1],   <-- modified
-           [0, 1, 0],
-           [0, 1, 1]]]]
-        ```
         """
-        # fmt: on
-
-        # Get the index of the first non-zero value for every sample in the batch.
-        # In the above example, indices = [[2], [0], [1]]]
-        tmp = torch.arange(attention_mask.shape[1], 0, -1)
-        indices = torch.argmax(attention_mask.cpu() * tmp, 1, keepdim=True)
-
-        # Find the batch indexes that have unattended tokens on the leftmost side (e.g. [0, 0, 1, 1, 1]), for which the first rows of the
-        # expanded mask will be completely unattended.
-        left_masked_rows = torch.where(indices > 0)[0]
-
-        if left_masked_rows.shape[0] == 0:
-            return expanded_mask
-        indices = indices[left_masked_rows]
-
-        max_len = torch.max(indices)
-        range_tensor = torch.arange(max_len).unsqueeze(0)
-        range_tensor = range_tensor.repeat(indices.size(0), 1)
-
-        # Avoid unmasking tokens at relevant target positions (on the row axis), by rather unmasking possibly several times the first row that should always be unmasked as we filtered out the batch above.
-        range_tensor[range_tensor >= indices] = 0
-
-        # TODO: we may drop support for 3D attention mask as the refactor from Patrick maybe dropped this case
-        if expanded_mask.dim() == 4:
-            num_masks = expanded_mask.shape[1]
-            if num_masks == 1:
-                # Broadcast [left_masked_rows, 1], [left_masked_rows, max_len]
-                mask_slice = (left_masked_rows[:, None], 0, range_tensor)
-            else:
-                # Broadcast [left_masked_rows, 1, 1], [1, num_masks, 1], [left_masked_rows, 1, max_len]
-                mask_slice = (
-                    left_masked_rows[:, None, None],
-                    torch.arange(num_masks)[None, :, None],
-                    range_tensor[:, None, :],
-                )
-        else:
-            # Broadcast [left_masked_rows, 1], [left_masked_rows, max_len]
-            mask_slice = (left_masked_rows[:, None], range_tensor)
-
-        expanded_mask[mask_slice] = unmasked_value
+        if expanded_mask.dtype == torch.bool:
+            raise ValueError("AttentionMaskConverter._unmask_unattended expects a float `expanded_mask`, got a BoolTensor.")
```

Comment:
Some models (gpt bigcode) use bool tensors, but Arthur's implem can't work for that dtype.

Reply:
can't we cast it and replace the `min` with 0?